cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -0,0 +1,337 @@
+	}
+
+	// If there are no vulnerabilities we can return.
+	if len(results) < 1 {
+		return nil
+	}
+
+	ap := report.ResourcesGroup{
+		Name: "Affected Packages",
+		Header: []string{
+			"Name",
+			"Version",
+			"Severity",
+			"FixedBy",
+		},
+	}
+
+	vp := report.ResourcesGroup{
+		Name: "Package Vulnerabilities",
+		Header: []string{
+			"Name",
+			"Version",
+			"Vulnerabilities",
+		},
+	}
+
+	var rows []map[string]string
+	duppedPackageVulns := make(map[string]map[string]string)
+	// As we are scanning only one container per check, we have only one item
+	// in results array.
+	for _, feature := range results[0].Vulnerabilities {
+		// Set global score for the report.
+		score := getScore(feature.Severity)
+		if score > vuln.Score {
+			vuln.Score = score
+		}
+
+		ap := map[string]string{
+			"Name":            feature.PkgName,
+			"Version":         feature.InstalledVersion,
+			"Severity":        feature.Severity,
+			"FixedBy":         feature.FixedVersion,
+			"Vulnerabilities": fmt.Sprintf("%s", feature.Description),
+		}
+		if len(ap["Vulnerabilities"]) > 512 {
+			reportTruncated = true
+			ap["Vulnerabilities"] = fmt.Sprintf("%s ...truncated", ap["Vulnerabilities"][0:500])
+		}
+
+		// Check if affected package has already been indexed.
+		if _, exists := duppedPackageVulns[ap["Name"]]; !exists {
+			duppedPackageVulns[ap["Name"]] = ap
+			continue
+		}
+
+		key := duppedPackageVulns[ap["Name"]]
+		if isMoreSevere(ap["Severity"], key["Severity"]) {
+			key["Severity"] = ap["Severity"]
+			// Only showing most sever vulnerability description to avoid
+			// report size overflow.
+			if ap["Vulnerabilities"] != "" {
+				key["Vulnerabilities"] = ap["Vulnerabilities"]
+			}
+		}
+		if version.Compare(version.Normalize(ap["FixedBy"]), version.Normalize(key["FixedBy"]), ">") {
+			key["FixedBy"] = ap["FixedBy"]
+		}
+
+		duppedPackageVulns[ap["Name"]] = key
+	}
+
+	for _, v := range duppedPackageVulns {
+		rows = append(rows, v)
+	}
+
+	// Sort rows by severity, alphabetical order of the package name and version.
+	sort.Slice(rows, func(i, j int) bool {
+		si := getScore(rows[i]["Severity"])
+		sj := getScore(rows[j]["Severity"])
+		switch {
+		case si != sj:
+			return si > sj
+		case rows[i]["Name"] != rows[j]["Name"]:
+			return rows[i]["Name"] < rows[j]["Name"]
+		default:
+			return rows[i]["Version"] < rows[j]["Version"]
+		}
+	})
+
+	// To avoid report size overflow only top 30 vulnerabilities are shown.
+	// In addition, docker commands are provided to let the user generate
+	// the full report.
+	totalVulnerablePackages := len(rows)
+	if totalVulnerablePackages > 30 {
+		log.Print("warning: truncate to top 30 vulnerabilities\n")
+		reportTruncated = true
+		rows = rows[0:30]
+	}
+
+	if reportTruncated {
+		vuln.Details = generateDetails(len(rows), totalVulnerablePackages, registryEnvDomain, target)
+	}
+
+	ap.Rows = rows
+	vp.Rows = rows
+
+	vuln.Resources = append(vuln.Resources, ap, vp)
+	state.AddVulnerabilities(vuln)
+
```

Comment:
Maybe it's good for debugging to store the report.json in the `report.Data` field.

Reply:
Unfortunately this is tricky. Depending on the size of the report.json the Vulcan report overflows and the check is reported as failed.